src/PowerShell/Utilities/PowerShellToolRunner.ts

```diff
@@ -0,0 +1,16 @@
+import * as io from '@actions/io';
+import * as exec from '@actions/exec';
+
+export default class PowerShellToolRunner {
+    static psPath: string;
+
+    static async init() {
+        if(!PowerShellToolRunner.psPath) {
+            PowerShellToolRunner.psPath = await io.which("pwsh", true);
+        }
+    }
+
+    static async executePowerShellCommand(command: string, options: any = {}) {
+        await exec.exec(`"${PowerShellToolRunner.psPath}" -Command "${command}"`, [], options);
```

Comment:
Can you do error handling for az cli login and ps login separately. Bubbling everything to common catch block in main.ts might be a little confusing.

Reply:
why have a separate catch block, since this catch block just logs the error? This is the same behavior we need in case of Azure PowerShell Login also right? If there is a specific error logic for both logins, maybe we can separate it out.

Reply:
I think having explicit messages like : failed creating az cli session or az ps session, gives customer more insight into what failed. Hence they would try local scripts before they open up a issue.

Reply:
In that case we need to use custom exceptions for each login to figure out which failed.
Please refer - https://stackoverflow.com/questions/3239906/multiple-try-catch-or-one

Reply:
set a flag after az cli succeeds. Use the value of that variable to differentiate between az cli failure and az ps failure.